web/src/user_group_edit.ts

```diff
@@ -1543,28 +1545,62 @@ export function maybe_reset_right_panel(groups_list_data: UserGroup[]): void {
 export function update_empty_left_panel_message(): void {
     // Check if we have any groups in panel to decide whether to
     // display a notice.
-    let has_groups;
     const is_your_groups_tab_active =
         get_active_data().$tabs.first().attr("data-tab-key") === "your-groups";
-    if (is_your_groups_tab_active) {
-        has_groups = user_groups.get_user_groups_of_user(people.my_current_user_id()).length;
-    } else {
-        has_groups = user_groups.get_realm_user_groups().length;
+
+    let current_group_filter =
+        z.string().optional().parse(filters_dropdown_widget.value()) ??
+        FILTERS.ACTIVE_AND_DEACTIVATED_GROUPS;
+
+    // When the dropdown menu is hidden.
+    if ($("#user-group-edit-filter-options").is(":hidden")) {
+        current_group_filter = FILTERS.ACTIVE_AND_DEACTIVATED_GROUPS;
     }
-    if (has_groups) {
+
+    const hidden_groups = $(".user-groups-list").children(":hidden").length;
+    const total_groups = $(".user-groups-list").children().length;
+
+    if (total_groups - hidden_groups > 0) {
```

Comment:
A simpler condition here would be `$(".user-groups-list").find(".group-row:visible").length > 0`.